src/components/structures/RoomSubList.js

```diff
@@ -151,10 +152,11 @@ var RoomSubList = React.createClass({
         }
     },
 
-    onRoomTileClick(roomId) {
+    onRoomTileClick(roomId, ev) {
         dis.dispatch({
             action: 'view_room',
             room_id: roomId,
+            clear_search: (ev && (ev.keyCode == KeyCode.ENTER || ev.keyCode == KeyCode.SPACE)),
```

Comment:
Oh, I think I see what's going on - so we rely on you tabbing to the RoomTile and hitting enter rather than the search box handling the enter, in which case this makes sense.